File: estrategia_bitcoin.php
<?php
header('Content-Type: application/json');

// 🔧 Função para limpar e validar dezenas
function limparDezenas($lista) {
  $limpas = [];
  foreach ($lista as $item) {
    $n = preg_replace('/\D/', '', $item);
    $n = intval($n);
    if ($n >= 1 && $n <= 25 && !in_array($n, $limpas)) {
      $limpas[] = $n;
    }
  }
  return $limpas;
}

// 📊 Funções auxiliares
function calcularMoldura($numeros) {
  $moldura = [1, 2, 3, 5, 6, 10, 11, 15, 16, 20, 21, 23, 24, 25];
  return count(array_intersect($numeros, $moldura));
}

function contarPares($numeros) {
  return count(array_filter($numeros, fn($n) => $n % 2 === 0));
}

function calcularMediaFrequencia($numeros, $frequencia) {
  $total = array_sum(array_map(fn($n) => $frequencia[$n] ?? 0, $numeros));
  return round($total / count($numeros), 2);
}

function gerarTrios($numeros) {
  $trios = [];
  for ($i = 0; $i < count($numeros) - 2; $i++) {
    $trios[] = "{$numeros[$i]}-{$numeros[$i+1]}-{$numeros[$i+2]}";
  }
  return $trios;
}

// 🔍 Função para gerar um jogo estratégico
function gerarJogo($maisFrequentes, $dezenasUltimo, $frequencia) {
  shuffle($maisFrequentes);
  $selecionados = array_slice($maisFrequentes, 0, 10);

  shuffle($dezenasUltimo);
  $ultimosSelecionados = array_slice($dezenasUltimo, 0, 5);

  $jogoBruto = array_merge($selecionados, $ultimosSelecionados);
  $jogo = limparDezenas($jogoBruto);
  $jogo = array_unique($jogo);
  sort($jogo);

  // Completa se necessário
  if (count($jogo) < 15) {
    foreach ($maisFrequentes as $n) {
      if (!in_array($n, $jogo)) {
        $jogo[] = $n;
        if (count($jogo) === 15) break;
      }
    }
  }

  if (count($jogo) !== 15) return null;

  // Estatísticas
  $dados = [
    "numeros" => $jogo,
    "repetidos" => array_values(array_intersect($jogo, $dezenasUltimo)),
    "frequentes" => array_values(array_intersect($jogo, array_slice($maisFrequentes, 0, 20))),
    "moldura" => calcularMoldura($jogo),
    "pares" => contarPares($jogo),
    "soma" => array_sum($jogo),
    "mediaFreq" => calcularMediaFrequencia($jogo, $frequencia),
    "trios" => gerarTrios($jogo),
    "aprovado" => true,
    "motivos" => []
  ];

  // 🧪 Peneira estatística
  if ($dados["moldura"] < 8 || $dados["moldura"] > 12) {
    $dados["aprovado"] = false;
    $dados["motivos"][] = "Moldura fora do padrão";
  }
  if ($dados["pares"] < 5 || $dados["pares"] > 10) {
    $dados["aprovado"] = false;
    $dados["motivos"][] = "Distribuição de pares desequilibrada";
  }
  if ($dados["soma"] < 170 || $dados["soma"] > 230) {
    $dados["aprovado"] = false;
    $dados["motivos"][] = "Soma fora da faixa ideal";
  }
  if ($dados["mediaFreq"] < 10) {
    $dados["aprovado"] = false;
    $dados["motivos"][] = "Média de frequência baixa";
  }

  return $dados;
}

// 📂 Carrega o histórico
$arquivo = 'lotofacil_combinacoes_convertido.json';
if (!file_exists($arquivo)) {
  echo json_encode(["erro" => "Arquivo de concursos não encontrado."]);
  exit;
}

$conteudo = file_get_contents($arquivo);
$concursos = json_decode($conteudo, true);
if (!$concursos || !is_array($concursos)) {
  echo json_encode(["erro" => "Erro ao ler os concursos."]);
  exit;
}

// 🔍 Último concurso e frequência
$chaves = array_keys($concursos);
$ultimo = max($chaves);
$dezenasUltimo = limparDezenas($concursos[$ultimo]);

$frequencia = [];
foreach ($concursos as $dezenas) {
  $limpas = limparDezenas($dezenas);
  foreach ($limpas as $n) {
    $frequencia[$n] = ($frequencia[$n] ?? 0) + 1;
  }
}
arsort($frequencia);
$maisFrequentes = array_keys($frequencia);

// 🔁 Gera múltiplos jogos
$jogos = [];
$tentativas = 0;
while (count($jogos) < 10 && $tentativas < 30) {
  $jogo = gerarJogo($maisFrequentes, $dezenasUltimo, $frequencia);
  if ($jogo && $jogo["aprovado"]) {
    $jogos[] = $jogo;
  }
  $tentativas++;
}

echo json_encode(["jogos" => $jogos]);
?>

File: estrategia_bitcoin.py
import json

def carregar_jogos_estrategicos(caminho_json):
    try:
        with open(caminho_json, 'r', encoding='utf-8') as arquivo:
            dados = json.load(arquivo)
            return dados.get("jogos", [])
    except Exception as e:
        print(f"❌ Erro ao carregar o arquivo: {e}")
        return []

def analisar_jogo(jogo, index):
    print(f"\n🎲 Jogo {index + 1}: {jogo['numeros']}")
    print(f"🔁 Repetidos do último concurso: {jogo.get('repetidos', [])}")
    print(f"📈 Frequentes: {jogo.get('frequentes', [])}")
    print(f"📊 Moldura: {jogo.get('moldura')}")
    print(f"⚖️ Pares: {jogo.get('pares')}")
    print(f"➕ Soma: {jogo.get('soma')}")
    print(f"📉 Média de frequência: {jogo.get('mediaFreq')}")
    print(f"📡 Trios orbitais: {', '.join(jogo.get('trios', [])[:5]) or 'Nenhum'}")
    
    if jogo.get("aprovado"):
        print("🧪 Peneira estatística: ✅ Aprovado")
    else:
        print("🧪 Peneira estatística: ❌ Reprovado")
        for motivo in jogo.get("motivos", []):
            print(f"   • {motivo}")

def executar_analise(caminho_json):
    jogos = carregar_jogos_estrategicos(caminho_json)
    if not jogos:
        print("⚠️ Nenhum jogo encontrado ou arquivo inválido.")
        return

    print(f"✅ {len(jogos)} jogos estratégicos carregados.\n")
    for i, jogo in enumerate(jogos):
        analisar_jogo(jogo, i)

# 🔧 Caminho do arquivo JSON gerado pelo PHP
executar_analise("estrategia_bitcoin.json")